src/test/java/com/google/devtools/build/lib/rules/cpp/Cpp20ModulesConfiguredTargetTest.java

```diff
@@ -0,0 +1,111 @@
+package com.google.devtools.build.lib.rules.cpp;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.devtools.build.lib.analysis.ConfiguredRuleClassProvider;
+import com.google.devtools.build.lib.analysis.util.AnalysisMock;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.analysis.util.DummyTestFragment;
+import com.google.devtools.build.lib.packages.util.Crosstool;
+import com.google.devtools.build.lib.testutil.TestRuleClassProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class Cpp20ModulesConfiguredTargetTest extends BuildViewTestCase {
+
+  @Override
+  protected ConfiguredRuleClassProvider createRuleClassProvider() {
+    ConfiguredRuleClassProvider.Builder builder = new ConfiguredRuleClassProvider.Builder();
+    TestRuleClassProvider.addStandardRules(builder);
+    builder.addConfigurationFragment(DummyTestFragment.class);
+    return builder.addRuleDefinition(new TestRuleClassProvider.MakeVariableTesterRule()).build();
+  }
+
+  @Test
+  public void testCpp20ModulesConfigurationNoFlags() throws Exception {
+    ImmutableList<String> targetList = ImmutableList.of("//foo:lib", "//foo:bin", "//foo:test");
+    scratch.file(
+        "foo/BUILD",
+        """
```

Comment:
I'd propose to move this `scratch.file` into a setup method, i.e.
```
@Before
public void setupBasicRulesWithModules() {
  scratch.file(....);
}
```

DRY is usually an antipattern for the tests, but in this case I'd recommend it.

supernit: fix weird indentation, make cc_library line up with opening """.